### PR TITLE
V10.6.1 private

### PR DESCRIPTION
**Pull request by adreed-msft**
Do not merge.

This is for a CI run.

### CODE REVIEW

File: cmd/copy.go [hunk 1]
```diff
@@ -1480,16 +1496,28 @@ func init() {
...
+				err := userFromTo.Parse(raw.fromTo)
+				if err != nil || (userFromTo != common.EFromTo.PipeBlob() && userFromTo != common.EFromTo.BlobPipe()) {
+					return fmt.Errorf("fatal: invalid from-to argument passed: %s", raw.fromTo)
+				}
+
+				if userFromTo == common.EFromTo.PipeBlob() {
+					// Case 1: PipeBlob. Check for the std input pipe
+					stdinPipeIn, err := isStdinPipeIn()
+					if stdinPipeIn == false || err != nil {
+						return fmt.Errorf("fatal: failed to read from Stdin due to error: %s", err)
```
Comment (zezha-msft) [inline cmd/copy.go:1513]: @mohsha-msft just to confirm, the error thrown here is on STDERR, correct?

File: cmd/sync.go [hunk 2]
```diff
@@ -533,6 +533,17 @@ func (cca *cookedSyncCmdArgs) process() (err error) {
 		return err
 	}
 
+	srcCredInfo, _, err := getCredentialInfoForLocation(ctx, cca.fromTo.From(), cca.source.Value, cca.source.SAS, true)
+
+	if err != nil {
+		return err
+	}
+
+	// If the source wants OAuth and the destination doesn't, override the credential type because this could be a download, or oauth to SAS.
```
Comment (zezha-msft) [inline cmd/sync.go:542]: A bit confused here, how can we do Oauth -> SAS sync? (I don't think we ever merged in the identity SAS change, right?)

File: cmd/helpMessages.go [hunk 3]
```diff
@@ -59,7 +59,11 @@ Upload a single file by using a SAS token:
 
 Upload a single file by using a SAS token and piping (block blobs only):
   
-  - cat "/path/to/file.txt" | azcopy cp "https://[account].blob.core.windows.net/[container]/[path/to/blob]?[SAS]"
+  - cat "/path/to/file.txt" | azcopy cp "https://[account].blob.core.windows.net/[container]/[path/to/blob]?[SAS]" --from-to PipeBlob
+
+Upload a single file by using OAuth and piping (block blobs only):
+
```
Comment (zezha-msft) [inline cmd/helpMessages.go:65]: I'm guessing that the example needs a login step?

**Review by zezha-msft — APPROVED** [review]
LGTM.

The CI may need some work to get passing.

Please also update the release note and version number.

File: cmd/copy.go (hunk 1 above)
Comment (mohsha-msft) [reply cmd/copy.go:1513]: yes correct.

**Comment by mohsha-msft** [comment]
> LGTM.
> 
> The CI may need some work to get passing.
> 
> Please also update the release note and version number.

@zezha-msft  https://github.com/Azure/azure-storage-azcopy/pull/1162/files Changes by Narasimha needs to be in it.

**Comment by zezha-msft** [comment]
Ok in that case this PR has go through the dev branch then, how else can we publish a release without having the tests run.

File: cmd/sync.go (hunk 2 above)
Comment (zezha-msft) [reply cmd/sync.go:542]: @adreed-msft

**Pull request closed without merging**